cryostatio/cryostat-web
# fix(demo): mirage demo fixups for 3.0

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Related to https://github.com/cryostatio/cryostatio.github.io/issues/176

## Description of the change:
- fixes one small potential bug in how the client requests report generation - the `reportUrl` will be prefixed by the server's authority if it is not already. This ensures that the client cannot be fooled into making a request to another server than the one it is hosted on. This also fixes a bug in the Mirage report generation
- fixes up a few small bugs and missing pieces in Mirage for the live demo

## Motivation for the change:
Preparing the live demo for 3.0.

## How to manually test:
1. `yarn start:dev:preview`
2. Most things vaguely work. You should be able to mostly click around the UI and not get any seriously broken features. Things that specifically don't work: editing recording labels, Archives > All Targets, and other GraphQL-powered features. Automated Rules can be created but don't actually do anything. Match Expressions, in automated rules and in stored credentials, always evaluate to `true`. But, you should be able to view Topology, Recordings, Events, and even create fake recordings on targets, view fake analysis reports on those recordings, stop/archive/delete recordings, etc.
